types/meteor-dburles-collection-helpers/mongo.d.ts

```diff
@@ -0,0 +1,88 @@
+    namespace Mongo {
+        interface Collection<T> {
+            /**
+             * Provide the definitions here for the methods and Helper<T>s you declared on your item interface.
+             * Tip: If you make those properties non-optional, they still won't be required when inserting items,
+             * but you'll know that any object of your interface type has them.
+             * Alternatively, if they're marked optional on your interface, they'll still be guaranteed on any
+             * object you get out of the collection.
+             * If you plan to pass around and create a lot of items pre-insertion, make them optional and use Full<T>
+             * for ones with helpers attached.
+             * If you plan to mostly pass around items that came out of a collection, make them required and use Data<T>
+             * when creating new items.
+             */
+            helpers(helpers: Helpers<T>): void;
+
+            // modifications:
+            // - replaced T with Full<T> & T everywhere Collection._transform is applied
+            // - replaced T with Data<T> everywhere the user provides a T
+
+            allow(options: {
+                insert?: (userId: string, doc: Full<T> & T) => boolean;
+                update?: (userId: string, doc: Full<T> & T, fieldNames: string[], modifier: any) => boolean;
+                remove?: (userId: string, doc: Full<T> & T) => boolean;
+                fetch?: string[];
+                // ditto
+                // tslint:disable-next-line ban-types
+                transform?: Function | null;
+            }): boolean;
+            deny(options: {
+                insert?: (userId: string, doc: Full<T> & T) => boolean;
+                update?: (userId: string, doc: Full<T> & T, fieldNames: string[], modifier: any) => boolean;
+                remove?: (userId: string, doc: Full<T> & T) => boolean;
+                fetch?: string[];
+                // ditto
+                // tslint:disable-next-line ban-types
+                transform?: Function | null;
+            }): boolean;
+            findOne(
+                selector?: Selector<T> | ObjectID | string,
+                options?: {
+                    sort?: SortSpecifier;
+                    skip?: number;
+                    fields?: FieldSpecifier;
+                    reactive?: boolean;
+                    // ditto
+                    // tslint:disable-next-line ban-types
+                    transform?: Function | null;
+                },
+            ): (Full<T> & T) | undefined;
+            // ditto
+            // tslint:disable-next-line ban-types
+            insert(doc: OptionalId<Data<T>>, callback?: Function): string;
+            update(selector: Selector<T> | ObjectID | string, modifier: Modifier<Data<T>>, options?: {
+                multi?: boolean;
+                upsert?: boolean;
+                arrayFilters?: Array<{ [identifier: string]: any }>;
+				// ditto
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/meteor-dburles-collection-helpers/mongo.d.ts;
  Line: 67;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.